### PR TITLE
Add metadata to plot

### PR DESCRIPTION
**Pull request by AnishereMariam**
All pull requests must have an associated issue in the issue tracker. If there
isn't one, please go open an issue describing the defect, deficiency or desired
feature. You can read more about our issue and PR processes in the
[wiki](https://github.com/bokeh/bokeh/wiki/BEP-1:-Issues-and-PRs-management).

- [x] issues: ref: #11765 
- [x] tests added / passed
- [ ] release document entry (if new feature or API change)

### CODE REVIEW

File: examples/basic/data/linear_cmap.py [hunk 1]
```diff
@@ -9,10 +9,10 @@
 
 bins = hexbin(x, y, 0.1)
 
-p = figure(tools="", match_aspect=True, background_fill_color='#440154')
+p = figure(tools="", match_aspect=True, background_fill_color='#00204C')
 p.grid.visible = False
 
 p.hex_tile(q="q", r="r", size=0.1, line_color=None, source=bins,
-           fill_color=linear_cmap('counts', 'Viridis256', 0, max(bins.counts)))
+           fill_color=linear_cmap('counts', 'Cividis256', 0, max(bins.counts)))
```
Comment (bryevdv) [inline examples/basic/data/linear_cmap.py:16]: I'd ask to undo the changes to this file so that the PR is focused on just the metadata changes

File: examples/plotting/customjs_expr.py [hunk 2]
```diff
@@ -1,3 +1,10 @@
+''' This example demonstrates using CustomJS Expression to create a sinusoidal line chart with Bokeh.
+
+.. bokeh-example-metadata::
+    :apis: bokeh.models.CustomJSExpr, bokeh.model.DataModel
+    :refs: :ref:`ug_interaction_callbacks_CustomJS`
```
Comment (bryevdv) [inline examples/plotting/customjs_expr.py]: needs to match the actual reference anchor in the sphinx docs, in this case:
```suggestion
    :refs: :ref:`ug_interaction_js_callback`
```

File: examples/plotting/filtering.py [hunk 3]
```diff
@@ -1,3 +1,16 @@
+''' A map representation of unemployment rate in US using the `US_Sates Dataset`_.
+This example demonstrates using IndexFilter, ColorMapper and HoverTool with 
+basic plot elements (`Area_glyphs.patches`). When hovering over the points,
```
Comment (bryevdv) [inline examples/plotting/filtering.py]:
```suggestion
basic plot elements such as patches. When hovering over the points, 
```
(FYI need *double* backticks for sphinx inline code formatting)

File: examples/plotting/filtering.py [hunk 4]
```diff
@@ -1,3 +1,16 @@
+''' A map representation of unemployment rate in US using the `US_Sates Dataset`_.
+This example demonstrates using IndexFilter, ColorMapper and HoverTool with 
+basic plot elements (`Area_glyphs.patches`). When hovering over the points, 
+the states and it's umemployment rate are shown.
```
Comment (bryevdv) [inline examples/plotting/filtering.py]: edits 
```suggestion
the state and its unemployment rate is shown. 
```

File: examples/plotting/filtering.py [hunk 5]
```diff
@@ -1,3 +1,16 @@
+''' A map representation of unemployment rate in US using the `US_Sates Dataset`_.
+This example demonstrates using IndexFilter, ColorMapper and HoverTool with 
+basic plot elements (`Area_glyphs.patches`). When hovering over the points, 
+the states and it's umemployment rate are shown. 
+
+.. bokeh-example-metadata::
+    :sampledata: US_States
+    :apis: bokeh.models.IndexFilter, bokeh.models.HoverTool, bokeh.models.LinearColorMapper
+    :refs: :ref:`ug_basic_patches` > :ref:`ug_basic_data_IndexFilter`
```
Comment (bryevdv) [inline examples/plotting/filtering.py]: correct references in this case (also separate with comma)
```suggestion
    :refs: :ref:`ug_basic_areas_patches`, :ref:`g_basic_data_filtering`
```

**Review by bryevdv — CHANGES_REQUESTED** [review]
a few small comments @anisheremariam

File: examples/plotting/filtering.py (hunk 3 above)
Comment (AnishereMariam) [reply examples/plotting/filtering.py]: Noted!

File: examples/basic/data/linear_cmap.py (hunk 1 above)
Comment (AnishereMariam) [reply examples/basic/data/linear_cmap.py:16]: I reverted this commit as requested and opened a new pull request for the task.

**Comment by AnishereMariam** [comment]
@bryevdv, I am sorry this is a bit messy. I am having a hard time creating separate PRs for each task I complete. Every commit I make keeps coming back to a single PR. I tried creating a separate branch for the other PR, but that did not work either. Does this mean I can not create multiple PRs?

**Comment by bryevdv** [comment]
> Does this mean I can not create multiple PRs?

Hi @anisheremariam you can definitely create multiple PRs from different branches. If you are having questions please PM me on Slack that will probably be the easiest place to have an async disussion

**Review by bryevdv — APPROVED** [review]

File: examples/plotting/filtering.py [hunk 6]
```diff
@@ -1,3 +1,16 @@
+''' A map representation of unemployment rate in US using the `US_Sates Dataset`_.
+This example demonstrates using IndexFilter, ColorMapper and HoverTool with 
+basic plot elements such as patches. When hovering over the points, 
+the state and its unemployment rate is shown.
```
Comment (bryevdv) [inline examples/plotting/filtering.py]:
```suggestion
This example demonstrates using IndexFilter, ColorMapper and HoverTool with
basic plot elements such as patches. When hovering over the points,
the state and its unemployment rate is shown.
```

**Review by bryevdv — CHANGES_REQUESTED** [review]
Last comment, there is some trailing space tripping up the linter

**Comment by bryevdv** [comment]
That said @anisheremariam this PR is almost ready to do (just one issue with trailing spaces)

**Comment by AnishereMariam** [comment]
@bryevdv, I have fixed all the changes you noted and found a solution to the Multiple PR issue. The community is a great help. Thank you.
